samples/createInstanceFromTemplateWithOverrides.js

```diff
@@ -0,0 +1,124 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * Creates GCP instance from an instance template with a new disk and machine type.
+ *
+ * @param {string} projectId - ID or number of the project you want to use.
+ * @param {string} zone - Name of the zone you want to check, for example: us-west3-b
+ * @param {string} instanceName - Name of the new instance.
+ * @param {string} instanceTemplateName - Name of the instance template using for creating the new instance.
+ * @param {string} machineType - Machine type you want to set in following format:
+ *    "zones/{zone}/machineTypes/{type_name}". For example:
+ *    "zones/europe-west3-c/machineTypes/f1-micro"
+ *    You can find the list of available machine types using:
+ *    https://cloud.google.com/sdk/gcloud/reference/compute/machine-types/list
+ * @param {string} newDiskSourceImage - Path the the disk image you want to use for your new
+ *    disk. This can be one of the public images
+ *    (e.g. "projects/debian-cloud/global/images/family/debian-10")
+ *    or a private image you have access to.
+ *    You can check the list of available public images using:
+ *    $ gcloud compute images list
+ */
+function main(
+  projectId,
+  zone,
+  instanceName,
+  instanceTemplateName,
+  machineType = 'n1-standard-2',
+  newDiskSourceImage = 'projects/debian-cloud/global/images/family/debian-10'
+) {
+  // [START compute_instances_create_from_template_with_overrides]
+  /**
+   * TODO(developer): Uncomment and replace these variables before running the sample.
+   */
+  // const projectId = 'YOUR_PROJECT_ID';
+  // const zone = 'europe-central2-b';
+  // const instanceName = 'YOUR_INSTANCE_NAME';
+  // const instanceTemplateName = 'YOUR_INSTANCE_TEMPLATE_NAME';
+  // const machineType = 'n1-standard-1';
+  // const newDiskSourceImage = 'projects/debian-cloud/global/images/family/debian-10';
+
+  const compute = require('@google-cloud/compute');
+  const computeProtos = compute.protos.google.cloud.compute.v1;
+
+  // Create a new instance from template in the specified project and zone with set machineType and new disk
+  async function createInstanceFromTemplateWithOverrides() {
+    const instancesClient = new compute.InstancesClient();
+    const instanceTemplatesClient = new compute.InstanceTemplatesClient();
+
+    console.log(
+      `Creating the ${instanceName} instance in ${zone} from template ${instanceTemplateName}...`
+    );
+
+    // Receiving instance template
+    const [instanceTemplate] = await instanceTemplatesClient.get({
+      project: projectId,
+      instanceTemplate: instanceTemplateName,
+    });
+
+    // Modifying diskTypes, cause diskType for instance insert function should be URL-formatted with specific zone.
```

Comment:
Replace comment with something like this for better readability:

> Adjust diskType field of the instance template to use the URL formatting required by instances.insert.diskType

Please verify accuracy